torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -21,18 +28,49 @@ namespace executor_utils {
 
 std::string kernelPreamble() {
   std::stringstream ss;
-  ss << code_template_tensor_struct << "\n"
-     << code_fp16_support << "\n"
-     << code_random_number_gen << "\n"
-     << code_helper_funcs << "\n"
-     << code_template_block_reduction << "\n"
-     << code_template_grid_reduction << "\n"
-     << code_template_block_broadcast << "\n";
+
+#ifndef __HIP_PLATFORM_HCC__
+  ss << R"(
+    #include <nvfuser_runtime/fp16_support.cu>
+  )";
+#endif
+
+  // clang-format off
+  ss << R"(
+    #include <nvfuser_runtime/tensor.cu>
+    #include <nvfuser_runtime/random_numbers.cu>
+    #include <nvfuser_runtime/helpers.cu>
+    #include <nvfuser_runtime/block_reduction.cu>
+    #include <nvfuser_runtime/grid_reduction.cu>
+    #include <nvfuser_runtime/broadcast.cu>
+  )";
+  // clang-format on
+
   return ss.str();
 }
 
 namespace {
 
+std::string torchLibPath() {
+#ifdef _WIN32
+// TODO $$$
+#else
+  // Used to lookup our library using dladdr()
+  static bool reference_symbol = false;
+
+  Dl_info dl_info = {};
+  TORCH_INTERNAL_ASSERT(dladdr(&reference_symbol, &dl_info));
+  const char* lib_filename = dl_info.dli_fname;
+
+  // TODO: a much better solution would be to use C++17's std::filesystem::path,
+  //  although that's currently not available to us (in the PyTorch codebase)
```

Comment:
Might be worth to try `<filesystem>` with their CI, C++17 features should be reasonable these days.

Reply:
Unfortunately the build system explicitly passes in std=c++14. This means that even if the toolchain supports newer features they can't be used.